pageserver/src/tenant/layer_map.rs

```diff
@@ -674,4 +710,16 @@ where
         println!("End dump LayerMap");
         Ok(())
     }
+
+    #[inline(always)]
+    fn compare_arced_layers(left: &Arc<L>, right: &Arc<L>) -> bool {
+        // FIXME: ptr_eq might fail to return true for 'dyn' references because of multiple vtables
+        // can be created in compilation. Clippy complains about this. In practice it seems to
+        // work, the assertion below would be triggered otherwise but this ought to be fixed.
```

Comment:
the assertion below is gone

Reply:
the assert should still be in place, but it's about l0 removal removing one, ot about this comparison, unless i misunderstand which assert?

Reply:
aah the comment was left unupdated, fixing

Reply:
[d4742ed](https://github.com/neondatabase/neon/pull/3513/commits/d4742edae00dce4ff71b02784ba4440b0b3b28c0)